posix-acl*: remove some calls to posix_acl_ctx_new() and cleanups

mykaul opened the pull request:
1. Remove unused 'this' parameter from functions.
2. posix_acl_destroy() was actually just a call to GF_FREE(acl), so replaced.
3. Removed unneeded calls to posix_acl_to_xattr() which were just to get the size
of the structs with the right call - a call to posix_acl_xattr_size()
4. posix_acl_new() would used to take a lock to increment the ref count of the acl
object it just created - for no reason.
5. Both posix_acl_lookup_cbk() and posix_acl_readdirp_cbk() used to call posix_acl_ctx_new(),
but do nothing with the resulting ctx. The following function, posix_acl_get() did it anyway,
so removed the redundant call.

Updates: #XXXX
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
posix-acl*: cleanups:

1. Remove unused 'this' parameter from functions.
2. posix_acl_destroy() was actually just a call to GF_FREE(acl), so replaced.
3. Removed unneeded calls to posix_acl_to_xattr() which were just to get the size
of the structs with the right call - a call to posix_acl_xattr_size()
4. posix_acl_new() would used to take a lock to increment the ref count of the acl
object it just created - for no reason.
5. Both posix_acl_lookup_cbk() and posix_acl_readdirp_cbk() used to call posix_acl_ctx_new(),
but do nothing with the resulting ctx. The following function, posix_acl_get() did it anyway,
so removed the redundant call.

Updates: #XXXX
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
DNM - need to understand if the calls I've removed were OK to remove.

mykaul commented:
/run regression

mykaul commented:
If someone can confirm the following is OK, then I believe it should be reviewed for merge:
```
Both posix_acl_lookup_cbk() and posix_acl_readdirp_cbk() used to call posix_acl_ctx_new(),
but do nothing with the resulting ctx. The following function, posix_acl_get() did it anyway,
so removed the redundant call.
```

mohit84 commented:
> If someone can confirm the following is OK, then I believe it should be reviewed for merge:
> 
> ```
> Both posix_acl_lookup_cbk() and posix_acl_readdirp_cbk() used to call posix_acl_ctx_new(),
> but do nothing with the resulting ctx. The following function, posix_acl_get() did it anyway,
> so removed the redundant call.
> ```

posix_acl_get does not create ctx, the function(posix_acl_ctx_get) has passed the _gf_false for create argument so the function would not create a ctx.

mykaul commented:
Thanks - I knew something was fishy. Let me fix or revert that.

Commit:
- Fixed review comment - posix_acl_lookup_cbk() and posix_acl_readdirp_cbk() need posix_acl_ctx_new()
- Added to __posix_acl_set(), posix_acl_set(), posix_acl_get(), posix_acl_ctx_update()
an additional parameter - ctx - if it is not NULL, they can use it and
skip the need to search for it.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/georep-basic-dr-rsync-arbiter.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/georep-basic-dr-rsync-arbiter.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/1995/

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/acl/acl-inherit.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/georep-basic-dr-tarssh.t
./tests/basic/acl/acl-inherit.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2001/

mykaul commented:
> Thanks - I knew something was fishy. Let me fix or revert that.

Fixed, I hope.
